firedrake/__init__.py

```diff
@@ -29,38 +29,11 @@
     import atexit
     atexit.register(lambda: _main_event.end())
     del atexit
-
-_blas_lib_path = petsc.get_blas_library()
 del petsc
 
-# UFL Exprs come with a custom __del__ method, but we hold references
-# to them /everywhere/, some of which are circular (the Mesh object
-# holds a ufl.Domain that references the Mesh).  The Python2 GC
-# explicitly DOES NOT collect such reference cycles (even though it
-# can deal with normal cycles).  Quoth the documentation:
-#
-#     Objects that have __del__() methods and are part of a reference
-#     cycle cause the entire reference cycle to be uncollectable,
-#     including objects not necessarily in the cycle but reachable
-#     only from it.
-#
-# To get around this, since the default __del__ on Expr is just
-# "pass", we just remove the method from the definition of Expr.
-import ufl
-try:
-    del ufl.core.expr.Expr.__del__
-except AttributeError:
-    pass
-del ufl
```

Comment:
Do we not need this anymore?

Reply:
Nope. From https://stackoverflow.com/questions/41851098/how-does-del-interfere-with-garbage-collection:

> Note that I link to the Python 2 documentation. As of Python 3.4, this is no longer true and even circular dependencies as show in the example will be cleared, as [PEP 442 – Safe object finalization](https://www.python.org/dev/peps/pep-0442/) has been implemented:

Reply:
Ok great